### PR TITLE
Update SSR for React docs to reflect newest version of React Router

### PR DESCRIPTION
**Pull request by eschaefer**
The existing docs show an example using React Router v3, which has a pretty different API than v4. I just got this working for v4, so let me know if anything seems unclear.

### CODE REVIEW

File: source/server-side-rendering.md [hunk 1]
```diff
@@ -70,52 +70,91 @@ We'll see how to take your component tree and turn it into a string in the next
 Once you put that all together, you'll end up with initialization code that looks like this:
 
 ```js
+/* This example uses React Router v4, although it should work equally well with other
```
Comment (stubailo) [inline source/server-side-rendering.md]: Can we make this a `//` comment?

File: source/server-side-rendering.md [hunk 2]
```diff
@@ -70,52 +70,91 @@ We'll see how to take your component tree and turn it into a string in the next
...
+        <li>
+          <Link to="/">Home</Link>
+        </li>
+        <li>
+          <Link to="/another">Another page</Link>
+        </li>
+      </ul>
+    </nav>
+
+    /* New <Switch> behavior introduced in React Router v4
```
Comment (stubailo) [inline source/server-side-rendering.md]: This isn't valid JSX, right? A comment would need to be within `{ }`

File: source/server-side-rendering.md [hunk 3]
```diff
@@ -70,52 +70,91 @@ We'll see how to take your component tree and turn it into a string in the next
...
+    */
+    <Switch>
+      {routes.map(props => <Route {...props} />)}
+    </Switch>
+  </div>;
+
+export default App;
+
+```
+
```
Comment (stubailo) [inline source/server-side-rendering.md]: A few extra empty lines

File: source/server-side-rendering.md [hunk 4]
```diff
@@ -70,52 +70,91 @@ We'll see how to take your component tree and turn it into a string in the next
...
+
+
+You can check out the [GitHunt app's `ui/server.js`](https://github.com/apollographql/GitHunt-React/blob/master/ui/server.js) for a complete working example with React Router v3, which has a slightly different API.
```
Comment (stubailo) [inline source/server-side-rendering.md]: If you have some time, it would be awesome to update that app to RR 4!

**Comment by stubailo** [comment]
Thanks for this, sorry for the review delay! A few small comments.

File: source/server-side-rendering.md (hunk 4 above)
Comment (eschaefer) [inline source/server-side-rendering.md]: Yes, will attempt this right now 🤘

**Comment by eschaefer** [comment]
Thanks @stubailo - I made the updates, and additionally opened a PR to update the GitHunt-React example to React Router v4  https://github.com/apollographql/GitHunt-React/pull/243

**Comment by stubailo** [comment]
Awesome thank you!

**Pull request merged**